Add documentation to sinkbinding modes (#3101)
knative/docs: docs/eventing/sources/sinkbinding.md

Sink binding injects environment variables into the `PodTemplateSpec` of the event sink, so that the application code does not need to interact directly with the Kubernetes API to locate the event destination.

> The `SinkBinding` operates in one of two modes; `Inclusion` or `Exclusion`.  This can be set by modifying the `SINK_BINDING_SELECTION_MODE` of the `eventing-webhook` deployment accordingly.
> 
> The selection mode determines the default scope of the webhook. By default, the upstream webhook is set to `exclusion` mode, which means any `Namespace` that does **not** have the label `bindings.knative.dev/exclude: true` will be subject to mutation evalutation.  Conversely, if `SINK_BINDING_SELECTION_MODE` is set to `inclusion`, only the resources in a namespaces labelled with `bindings.knative.dev/include: true` will be considered.  In `inclusion` mode, any `sinkbinding` resource created will automatically label the `subject`'s namespace with `bindings.knative.dev/include: true` for inclusion in the potential environment variable inclusions.

Reviewer: ```suggestion
Sink binding operates in one of two modes; `Inclusion` or `Exclusion`.
You can set the mode by modifying the `SINK_BINDING_SELECTION_MODE` of the `eventing-webhook` deployment accordingly. The mode determines the default scope of the webhook.

By default, the webhook is set to `exclusion` mode, which means that any namespace that does not have the label `bindings.knative.dev/exclude: true` will be subject to mutation evalutation.

If `SINK_BINDING_SELECTION_MODE` is set to `inclusion`, only the resources in a namespace labelled with `bindings.knative.dev/include: true` will be considered.  In `inclusion` mode, any SinkBinding resource created will automatically label the `subject` namespace with `bindings.knative.dev/include: true` for inclusion in the potential environment variable inclusions.
```
